File: AAC-Practice/app/src/main/java/com/bowoon/android/aac_practice/MemoViewModel.java
package com.bowoon.android.aac_practice;

import android.app.Application;

import java.util.List;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MediatorLiveData;
import androidx.lifecycle.Observer;

/**
 * UI 관련 데이터를 관리하는 클래스
 */
public class MemoViewModel extends AndroidViewModel {
    private final DataRepository repository;
    private final MediatorLiveData<List<Memo>> observableMemo;

    public MemoViewModel(Application application) {
        super(application);

        observableMemo = new MediatorLiveData<>();
        observableMemo.setValue(null);

        repository = ((BasicApp) application).getRepository();
        LiveData<List<Memo>> memo = repository.getMemo();

        observableMemo.addSource(memo, new Observer<List<Memo>>() {
            @Override
            public void onChanged(List<Memo> memos) {
                observableMemo.setValue(memos);
            }
        });
    }

    public LiveData<List<Memo>> getAllMemo() {
        return observableMemo;
    }

    public void addMemo(Memo memo) {
        repository.addMemo(memo);
    }

    public void deleteMemo(Memo memo) {
        repository.deleteMemo(memo);
    }
}

File: AACPracticeKotlin/app/src/main/java/com/bowoon/android/aac_practice_kotlin/MemoViewModel.kt
package com.bowoon.android.aac_practice_kotlin

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MediatorLiveData
import androidx.lifecycle.Observer


class MemoViewModel(application: Application) : AndroidViewModel(application) {
    private var repository: DataRepository
    private var observableMemo: MediatorLiveData<List<Memo>> = MediatorLiveData()

    init {
        observableMemo.value = null
        repository = (application as BasicApp).getRepository()!!
        val memo = repository.memo
        observableMemo.addSource(memo, object : Observer<List<Memo>> {
            override fun onChanged(memos: List<Memo>) {
                observableMemo.value = memos
            }
        })
    }

    fun getAllMemo(): LiveData<List<Memo>> {
        return observableMemo
    }

    fun addMemo(memo: Memo) {
        repository.addMemo(memo)
    }

    fun deleteMemo(memo: Memo) {
        repository.deleteMemo(memo)
    }
}